argo_jupyter_scheduler/executor.py

```diff
@@ -369,7 +396,43 @@ def _create_cwf_oject(
                         "job_definition_id": job_definition_id,
                     },
                 )
+
                 Step(name="main", template=main, continue_on=ContinueOn(failed=True))
+
+                rename_files(
+                    name="rename-files",
+                    arguments={
+                        "db_url": db_url,
+                        "job_definition_id": job_definition_id,
+                        "input_path": input_path,
+                        "log_path": log_path,
+                        "start_time": None,
+                    },
+                    continue_on=ContinueOn(failed=True),
+                )
+
+                failure += " || {{steps.rename-files.status}} == Failed"
+                successful += " && {{steps.rename-files.status}} == Succeeded"
```

Comment:
Note for reviewers: the steps here run as part of the detachable cron-like job workflow. We don't know when a job will be scheduled until it's scheduled in the `create_job_record` step. The `start_time` value needs to be generated within that step. But we cannot pass the `start_time` value directly between these steps. Instead, we use the database.

When `rename_files` is called in the other (non-cron) workflow, we just pass `job.create_time` since in that case it would match the job start time. There's no `create_job_record` step in that case and the job is executed only once. The DB-related parameters are set to `None` in that case since they are not needed/used.

Getting the correct `start_time` is important since it needs to match what the upstream jupyter-scheduler extension expects. Otherwise, you won't be able to download output files via the UI when the job is done executing.

Other steps  using these arguments work similarly.

Note that `rename_files` is set to continue on failure. This is necessary for the later steps to get executed, which update the job status in the UI. `rename_files` is also deliberately executed regardless of whether the `main` step succeeded. This allows for downloading output files via the web UI if an exception is raised during notebook execution.

Reply:
>The start_time value needs to be generated within that step. But we cannot pass the start_time value directly between these steps. Instead, we use the database.

What do you mean by generated? Isn't `start_time` the time of start of the job and not some thing that's generated?

Reply:
Usually, yes. But not in this case. In `create_job_record`, we create a new job definition to have these detachable cron jobs and set this:

```py
        job.start_time = get_utc_timestamp()
```

What I was trying to say above is that this code (`get_utc_timestamp`) needs to run inside the `create_job_record` `script` and not somewhere within `with CronWorkflow` and not within `with Steps`.

If we had it within those two `with` blocks, it would have the same time value (when the workflow was defined initially) instead of being the time the pod was actually executed.

So having it inside the `script` is the only way to make sure it's executed every time whenever the pod runs. This is what we want because we want these times to be unique (corresponding to subsequent Notebook job executions).

LMK if this is still not clear.